lib/phoenix_component.ex

```diff
@@ -658,17 +659,31 @@ defmodule Phoenix.Component do
       compile_error!(line, file, "only one of :required or :default must be given")
     end
 
+    {slot, opts} = Keyword.pop(opts, :slot, nil)
+
     type = validate_attr_type!(module, name, type, line, file)
     validate_attr_opts!(name, opts, line, file)
 
-    Module.put_attribute(module, :__attrs__, %{
-      name: name,
-      type: type,
-      required: required,
-      opts: opts,
-      doc: doc,
-      line: line
-    })
+    if slot do
+      Module.put_attribute(module, :__slot_attrs__, %{
+        name: name,
+        type: type,
+        required: required,
+        opts: opts,
+        doc: doc,
+        line: line,
+        slot: slot
+      })
+    else
+      Module.put_attribute(module, :__attrs__, %{
+        name: name,
+        type: type,
+        required: required,
+        opts: opts,
+        doc: doc,
+        line: line
+      })
```

Comment:
I understand it is still draft, but in case you are accepting suggestions already 😅 

```elixir
    
attrs = %{
 name: name,
 type: type,
 required: required,
 opts: opts,
 doc: doc,
 line: line
}

if slot do
  Module.put_attribute(module, :__slot_attrs__, Map.put(attrs, :slot, slot) )
else
  Module.put_attribute(module, :__attrs__, attrs)
end
```

Reply:
Yep, this is very much a work-in-progress as I iron out the approach, apologies for the mess 🧹 😀